core/src/main/java/org/apache/accumulo/core/conf/cluster/ClusterConfigParser.java

```diff
@@ -87,6 +93,20 @@ private static void flatten(String parentKey, String key, Object value,
 
   public static void outputShellVariables(Map<String,String> config, PrintStream out) {
 
+    config.keySet().forEach(section -> {
+      if (VALID_CONFIG_KEYS.contains(section)) {
+        return;
+      }
+
+      for (String validPrefix : VALID_CONFIG_PREFIXES) {
+        if (section.startsWith(validPrefix)) {
+          return;
+        }
+      }
+
+      throw new IllegalArgumentException("Unknown configuration section : " + section);
+    });
```

Comment:
Here's the stream simplification to replace the forEach. I preserved behavior here, but this is a lot simpler if you're not trying to extract the invalid config section to put in the message. In that case, you could just do: `if (config.keySet().stream().anyMatch(VALID_CONFIG_SECTIONS.negate()) { throw ...` or negate the check instead of the Predicate, as in `if (!config.keySet().stream().allMatch(VALID_CONFIG_SECTIONS) { throw ...`

```suggestion
    // find invalid config sections and point the user to the first one
    config.keySet().stream().filter(VALID_CONFIG_SECTIONS.negate()).findFirst()
        .ifPresent(section -> {
          throw new IllegalArgumentException("Unknown configuration section : " + section);
        });
```

There are also variants of this that would collect the entire set of invalid sections and put them all in the error message, or to put each one into their own suppressed exception, reduced into a single one to throw. But, those are probably overkill.